int64[] row_data